### PR TITLE
Refactor microservice

### PR DESCRIPTION
**Pull request by sltsheryl**
# To set up 
- add `.env` in `question-service` (it was previously in `backend/src`)
- open docker app
- `cd backend/question-service`
- `docker build . -t question-service`
- `docker run -p 8080:8080 -d question-service` (the first port number is where it will run, so if you do `49160:8080`, it'll be `localhost:49160/api/questions`)

Shd see it running

Screenshot 2023-09-16 at 12 27 58 AM

This was supposed to be assignment 4, but I found this while reading more about microservices design while trying to change the backend dir.

### CODE REVIEW

File: backend/question-service/src/controllers/questions.ts
Comment (seelengxd) [inline backend/question-service/src/controllers/questions.ts:1]: i think this undoes changes from #27 and update does not work

**Review by seelengxd — COMMENTED** [review]
i like how you also dockerised the app and are always so fast :D

File: backend/question-service/src/controllers/questions.ts
Comment (sltsheryl) [reply backend/question-service/src/controllers/questions.ts:1]: oh yeah i didn't notice it got reverted, thks

**Review by seelengxd — APPROVED** [review]
LGTM! i like that you added a spinner as well in the indiv page :P

**Pull request merged**